genesis/engine/solvers/rigid/collider_decomp.py

```diff
@@ -86,7 +86,9 @@ def _init_static_config(self) -> None:
                 geom.is_convex or geom.type == gs.GEOM_TYPE.TERRAIN for geom in self._solver.geoms
             ),
             has_terrain=any(geom.type == gs.GEOM_TYPE.TERRAIN for geom in self._solver.geoms),
-            n_contacts_per_pair=20 if self._solver._static_rigid_sim_config.requires_grad else 5,
+            # FIXME: The line below will cause error for tests/test_grad.py::test_diff_solver[cpu], why?
```

Comment:
when you have time, can you take a look @SonSang

Reply:
The differentiable contact detection finds more than 5 contacts, so error should be thrown. And yes, you can go ahead. I will take care of this later again.

Reply:
I'm in favour of not breaking stuffs that work just to slightly reduce memory usage.